File: 202311/1980.go
func findDifferentBinaryString(nums []string) string {
    n := len(nums)
    var s []byte

    as := make(map[string]bool)
    for _, item := range nums {
        as[item] = true
    }

    var ans string

    var f func(k int)
    f = func(k int) {
        if len(ans) > 0 {
            return
        }
        if k == n {
            if _, ok := as[string(s)]; !ok {
                ans = string(s)
            }
            return
        }
        s = append(s, '0')
        f(k+1)
        s = s[0:len(s)-1]
        s = append(s, '1')
        f(k+1)a
        s = s[0:len(s)-1]
    }
    f(0)
    return ans
}

File: 202603/1980.py
class Solution:
    def findDifferentBinaryString(self, nums: List[str]) -> str:
        s = set(nums)
        n = len(nums[0])
        for i in range(1 << n):
            
            a = f"{i:0{n}b}"
            if a not in s:
                return a
        return ""
